readers/api/transport.go

```diff
@@ -142,7 +139,8 @@ func decodeList(ctx context.Context, r *http.Request) (interface{}, error) {
 
 	req := listMessagesReq{
 		chanID: bone.GetValue(r, "chanID"),
-		token:  r.Header.Get("Authorization"),
+		token:  apiutil.ExtractBearerToken(r),
+		key:    apiutil.ExtractThingKey(r),
```

Comment:
I don't like having both token and key in the list request. I prefer that `apiutil.ExtractAuthToken(r)` can decide what's the auth scheme and that auth gRPC client knows how to authorize different tokens. This can be done by either separating APIs or updating auth gRPC. Anyways, this is not related to this PR, and we'll create a separate issue from this comment to address it later and not to block this PR.

Reply:
@dusanb94 @manuio is the issue for this created? If not, can you please create it and paste the reference here.